src/main/java/teammates/ui/webapi/action/RegenerateStudentCourseLinksAction.java

```diff
@@ -0,0 +1,73 @@
+package teammates.ui.webapi.action;
+
+import org.apache.http.HttpStatus;
+
+import teammates.common.datatransfer.attributes.StudentAttributes;
+import teammates.common.exception.EntityDoesNotExistException;
+import teammates.common.exception.UnauthorizedAccessException;
+import teammates.common.util.Config;
+import teammates.common.util.Const;
+import teammates.common.util.EmailSendingStatus;
+import teammates.common.util.EmailWrapper;
+import teammates.common.util.StringHelper;
+import teammates.common.util.Templates;
+import teammates.ui.webapi.output.RegenerateStudentCourseLinksData;
+
+/**
+ * Regenerates the course join and feedback session links for a given student in a course.
+ */
+public class RegenerateStudentCourseLinksAction extends Action {
+
+    private static final String SUCCESSFUL_UPDATE = "Student's links for this course have been regenerated,";
+    private static final String SUCCESSFUL_UPDATE_WITH_EMAIL = SUCCESSFUL_UPDATE + " and the email has been sent.";
+    private static final String SUCCESSFUL_UPDATE_BUT_EMAIL_FAILED = SUCCESSFUL_UPDATE + " but the email failed to send.";
+
+    @Override
+    protected AuthType getMinAuthLevel() {
+        return AuthType.LOGGED_IN;
+    }
+
+    @Override
+    public void checkSpecificAccessControl() {
+        if (!Config.isDevServer()) {
+            throw new UnauthorizedAccessException("Admin privilege is required to access this resource.");
+        }
+    }
+
+    @Override
+    public ActionResult execute() {
+        String courseId = getNonNullRequestParamValue(Const.ParamsNames.COURSE_ID);
+        String studentEmailAddress = getNonNullRequestParamValue(Const.ParamsNames.STUDENT_EMAIL);
```

Comment:
check if this is really an email address

Reply:
Is it required? I don't see any of the other `Action`s doing this. 

The `FieldValidator#getInvalidityInfoForEmail` checks the email, and returns a detailed message if it is invalid when creating the new entity. So if this fails, there won't be an attempt to send the email anyways

Reply:
You can check out junming's session link recovery pr. Detailed reasons listed there.

Reply:
Are you referring to [this](https://github.com/TEAMMATES/teammates/pull/9413#discussion_r265462937). I don't mind adding it. It's just that it won't be consistent with the rest of the codebase's `Action`s. Would we have to update them as well?

The main difference with this PR and session link recovery is that the email in this case is not manually input, compared to session links recovery. Also, here, there are implicit checks like `logic.getStudentForEmail()` (which will return error if it's null)